documentation/src/main/asciidoc/topics/con_cache_store_jdbc_connection_factories.adoc

```diff
@@ -0,0 +1,65 @@
+[id='jdbc-connection-factories_{context}']
+= JDBC connection factories
+
+{brandname} provides different `ConnectionFactory` implementations that allow you to connect to databases.
+You use JDBC connections with SQL cache stores and JDBC string-based caches stores.
+
+This section provides examples with JDBC string-based stores but the same configuration applies to SQL cache stores.
+For example, in declarative configuration you can replace the `string-keyed-jdbc-store` element with either the `table-jdbc-store` or `query-jdbc-store` elements.
+
+[discrete]
+== Connection pools
+
+Connection pools are suitable for standalone {brandname} deployments and are based on Agroal.
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/persistence_jdbc_connection_pool.xml[]
+----
+
+.ConfigurationBuilder
+[source,java,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::code_examples/PersistenceJdbcConnectionPool.java[]
+----
+
+[discrete]
+== Managed datasources
+
+Datasource connections are suitable for managed environments such as application servers.
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/persistence_jdbc_connection_datasource.xml[]
+----
+
+.ConfigurationBuilder
+[source,java,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::code_examples/PersistenceJdbcConnectionDatasource.java[]
+----
+
+[discrete]
+== Simple connections
+
+Simple connection factories create database connections on a per invocation basis and are intended for use with test or development environments only.
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/persistence_jdbc_connection_simple.xml[]
+----
+
+.ConfigurationBuilder
+[source,java,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::code_examples/PersistenceJdbcConnectionSimple.java[]
+----
+
+[role="_additional-resources"]
+.Additional resources
+* link:{javadocroot}/org/infinispan/persistence/jdbc/configuration/PooledConnectionFactoryConfigurationBuilder.html[PooledConnectionFactoryConfigurationBuilder]
+*  link:{javadocroot}/org/infinispan/persistence/jdbc/configuration/ManagedConnectionFactoryConfigurationBuilder.html[ManagedConnectionFactoryConfigurationBuilder]
+*  link:{javadocroot}/org/infinispan/persistence/jdbc/configuration/SimpleConnectionFactoryConfigurationBuilder.html[SimpleConnectionFactoryConfigurationBuilder]
```

Comment:
@wburns and @ryanemerson These Javadocs seem to have disappeared in 13. `jdbc/common/configuration` is not included.

Reply:
These classes moved to the PooledConnectionFactoryConfigurationBuilder package instead is why.

Reply:
@wburns Right. What I mean though is that I don't see anything under "javadoc/target/apidocs/org/infinispan/persistence/jdbc/common/" when I build the distribution.

Reply:
@wburns I've fixed it. The jdbc-common stuff was missing from the pom for the Javadocs. Also the SQL store package was not marked as public so it was not included either. 

I've made an update to the SQL store XSD as well because it didn't build as part of the configdocs in the distribution but I don't see RocksDB and some of the other stores in there either so further investigation might be needed.